opentelemetry-api/src/opentelemetry/trace/__init__.py

```diff
@@ -157,12 +167,25 @@ class SpanContext:
     def __init__(self,
                  trace_id: int,
                  span_id: int,
-                 options: 'TraceOptions',
-                 state: 'TraceState') -> None:
+                 traceoptions: 'TraceOptions' = None,
+                 tracestate: 'TraceState' = None
+                 ) -> None:
+        if traceoptions is None:
+            traceoptions = DEFAULT_TRACEOPTIONS
+        if tracestate is None:
+            tracestate = DEFAULT_TRACESTATE
         self.trace_id = trace_id
         self.span_id = span_id
-        self.options = options
-        self.state = state
+        self.traceoptions = traceoptions
+        self.tracestate = tracestate
+
+    def __repr__(self) -> str:
+        return ("{}(trace_id={}, span_id={})"
+                .format(
+                    type(self).__name__,
+                    format_trace_id(self.trace_id),
+                    format_span_id(self.span_id)
```

Comment:
Curious, do we omit traceoptions and tracestate here on purpose?

Reply:
Just for brevity, I'm assuming we don't want to include all fields in all reprs. Especially since classes like `Span` have so many fields as to make that kind of repr effectively useless.

It might be misleading that this produces a string that looks like a constructor call though, and making that actual call would produce a different object.

Reply:
Got it.